src/vsphere_cpi/lib/cloud/vsphere/vm_type.rb

```diff
@@ -15,7 +15,7 @@ def initialize(datacenter, cloud_properties, pbm)
     end
 
     %w[cpu ram disk cpu_hot_add_enabled memory_hot_add_enabled nsx vmx_options
-       nsxt nested_hardware_virtualization upgrade_hw_version datacenters vm_group disable_drs storage_policy tags].each do |name|
+       nsxt memory_reservation_locked_to_max nested_hardware_virtualization upgrade_hw_version datacenters vm_group disable_drs storage_policy tags].each do |name|
```

Comment:
Looks good.
I _believe_ that by setting this to true & attempting to power on the VM & if there's not enough memory (`cluster_free_memory` - `overall_usage`) it won't power on.

We should, at some point, update the bosh.io docs to reflect this new property.

From the vSphere docs:

> If set true, memory resource reservation for this virtual machine will always be equal to the virtual machine's memory size; increases in memory size will be rejected when a corresponding reservation increase is not possible. This feature may only be enabled if it is currently possible to reserve all of the virtual machine's memory

Reply:
Wat ahead of you 🙃 https://github.com/cloudfoundry/docs-bosh/pull/779